File: dev/test_data/main/golang/expected/implementation_specific/input/snippets/Verification/texts_are_unique.go
// Check that the `texts` do not repeat.
func TextsAreUnique(texts []string) bool {
	textSet := make(map[string]struct{})

	for _, text := range texts {
		_, has := textSet[text]
		if has {
			return false
		}

		textSet[text] = struct{}{}
	}
	return true
}

File: dev/test_data/main/typescript/expected/implementation_specific/input/snippets/Verification/texts_are_unique.ts
/**
 * Check that the `texts` do not repeat.
 *
 * @param texts - to be verified
 * @returns `true` if the check passes
 */
export function textsAreUnique(texts: Iterable<string>): boolean {
  const textSet = new Set<string>();
  for (const text of texts) {
    if (textSet.has(text)) {
      return false;
    }

    textSet.add(text);
  }

  return true;
}
